sentry-android-core/src/main/java/io/sentry/android/core/ViewHierarchyEventProcessor.java

```diff
@@ -0,0 +1,169 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import androidx.annotation.NonNull;
+import io.sentry.Attachment;
+import io.sentry.EventProcessor;
+import io.sentry.Hint;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.internal.gestures.ViewUtils;
+import io.sentry.protocol.ViewHierarchy;
+import io.sentry.protocol.ViewHierarchyNode;
+import io.sentry.util.Objects;
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** ViewHierarchyEventProcessor responsible for taking a snapshot of the current view hierarchy. */
+@ApiStatus.Internal
+public final class ViewHierarchyEventProcessor implements EventProcessor {
+
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  private static final @NotNull Charset UTF_8 = Charset.forName("UTF-8");
+
+  private static final long TIMEOUT_PROCESSING_MILLIS = 1000;
+
+  private final @NotNull SentryAndroidOptions options;
+
+  public ViewHierarchyEventProcessor(final @NotNull SentryAndroidOptions options) {
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+  }
+
+  @Override
+  public @NotNull SentryEvent process(final @NotNull SentryEvent event, final @NotNull Hint hint) {
+    if (!event.isErrored()) {
+      return event;
+    }
+
+    if (!options.isAttachViewHierarchy()) {
+      options.getLogger().log(SentryLevel.DEBUG, "attachViewHierarchy is disabled.");
+      return event;
+    }
+
+    final @Nullable Activity activity = CurrentActivityHolder.getInstance().getActivity();
+    if (activity == null) {
+      return event;
+    }
+
+    final @Nullable Window window = activity.getWindow();
+    if (window == null) {
+      return event;
+    }
+
+    final @Nullable View decorView = window.peekDecorView();
+    if (decorView == null) {
```

Comment:
`l`: Not sure if we should add log statements for all of the ifs above? Would be easier to investigate potential issues of why the view hierarchy hasn't been captured by just enabling the debug log

Reply:
Good point, I've added some logs - but at an INFO level as it could be completely fine that there's not activity or window active at the moment the crash happened.